aclk/mqtt.c

```diff
@@ -269,25 +276,11 @@ int _link_set_lwt(char *sub_topic, int qos)
         return 1;
     }
 
-    usec_t time_created_offset_usec = now_realtime_usec();
-    time_t time_created = time_created_offset_usec / USEC_PER_SEC;
-    time_created_offset_usec = time_created_offset_usec % USEC_PER_SEC;
-
-    char *msg_id = create_uuid();
-
-    snprintfz(
-        payload, 511,
-        "{ \"type\": \"disconnect\","
-        " \"msg-id\": \"%s\","
-        " \"timestamp\": %ld,"
-        " \"timestamp-offset-usec\": %llu,"
-        " \"version\": %d,"
-        " \"payload\": \"unexpected\" }",
-        msg_id, time_created, time_created_offset_usec, ACLK_VERSION);
-
-    freez(msg_id);
+    usec_t lwt_time = aclk_session_sec * USEC_PER_SEC + aclk_session_us + 1;
+    BUFFER *b = buffer_create(512);
+    aclk_create_header(b, "disconnect", NULL, lwt_time / USEC_PER_SEC, lwt_time % USEC_PER_SEC);
```

Comment:
We need to retain the unexpected/graceful disconnection indicator for anonymous statistics/metrics purposes. You don't seem to have updated the respective part in `aclk_main_cleanup()`. Don't you need to do that in order for the cloud backend to be able to match the session id safely?

Reply:
This needs to add the payload `unexpected` string and '}' for the json object

Reply:
Code wasn't finished, partial check-in while I had dinner. The new version should be complete but can't be tested until I get the local dev env up and running again.